lib/src/common.dart

```diff
@@ -111,4 +106,24 @@ class Common {
 
     return result;
   }
+
+  /// Gets or creates the cookie vault.
+  static Vault<String> getVault() {
+    return vault ?? (vault = createDefaultCookieVault());
+  }
+
+  /// Creates a default, plain-text Hive store and vault in the system temporary directory.
+  static Vault<String> createDefaultCookieVault() {
+    final path = Directory.systemTemp.path;
+    final store = newHiveDefaultVaultStore(path: path);
+    return store.vault<String>(
+        name: 'cookieVault', eventListenerMode: EventListenerMode.synchronous);
+  }
+
+  /// Defines the cookie vault for this process.
+  /// No-op if a cookie vault is already defined.
+  static Vault<String> setCookieVault(Vault<String> newVault) {
+    assert(vault == null, 'Cookie vault is already initialized');
+    return vault ?? (vault = newVault);
+  }
```

Comment:
```suggestion
  /// Gets or creates the cookie vault.
  static Vault<String> getVault() {
    return vault ?? (vault = createCookieVault());
  }

  /// Creates a Hive store and vault with the optional [path], [vaultName] and
  /// [encryptionCipher] parameters. By default will create a plain-text
  /// Hive store and vault in the system temporary directory.
  static Vault<String> createCookieVault({
    final String? path,
    final String? vaultName,
    final HiveCipher? encryptionCipher,
  }) {
    final String defaultPath = Directory.systemTemp.path;
    final String defaultVaultName = 'cookieVault';

    final store = newHiveDefaultVaultStore(
      path: path ?? defaultPath,
      encryptionCipher: encryptionCipher,
    );
    return store.vault<String>(
      name: vaultName ?? defaultVaultName,
      eventListenerMode: EventListenerMode.synchronous,
    );
  }

  /// Defines the cookie vault for this process with the optional [path], 
  /// [vaultName] and [encryptionCipher] parameters.
  /// No-op if a cookie vault is already defined.
  static void setCookieVault({
    final String? path,
    final String? vaultName,
    final HiveCipher? encryptionCipher,
  }) {
    assert(vault == null, 'Cookie vault is already initialized');

    if (vault != null) {
      vault = createCookieVault(
        path: path,
        vaultName: vaultName,
        encryptionCipher: encryptionCipher,
      );
    }
  }
```

Why not defining a general method for creating cookie vaults ?

There is no need for `setCookieVault` to return anything since it is a setter.

I also replaced the parameters to match a previous comment (https://github.com/jossef/requests/pull/63#issuecomment-1105316402).